### PR TITLE
Fix the issue #32: Annotation TopSupported should be a Tag, not a Record

### PR DESCRIPTION
**Pull request by xuzhg**
Add 'Location' property to allow setting the serialization location

### Issues

*This pull request fixes issue #32*

### Description

*Briefly describe the changes of this pull request.*

1)  Fix the issue #32: Annotation TopSupported should be a Tag, not a Record
2)  Add 'Location' property to allow setting the serialization location

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt [hunk 1]
```diff
@@ -390,6 +390,58 @@ namespace <#= @namespace #>
...
+            if (fullTypeName == "Edm.String")
+            {
+#>
+            return new EdmStringConstant(_<#= camelName #>);
+<#+
+            }
+            else if (fullTypeName == "Edm.PrimitiveType")
+            {
+#>
+            if (_<#= camelName #> != null)
```
Comment (g2mula) [inline src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt:408]: issue: Testing for null implies a primitive type can be null, if so, this method will not compile.
suggestion: Add an `else` for when `_<#= camelName #> == null`

Comment (xuzhg) [reply src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt:408]: #Resolved

**Comment by xuzhg** [comment]
> @xuzhg Update the public API baseline. At the same time consider whether this could be a breaking change...

From public API, it could be a public API change. But, for the functionality, it's a fix for our mistakes.

File: src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt [hunk 2]
```diff
@@ -390,6 +390,58 @@ namespace <#= @namespace #>
...
+            else if (fullTypeName == "Edm.Int16")
+            {
+#>
+            return new EdmIntegerConstant(_<#= camelName #> ?? 0);
+<#+
+            }
+            else if (fullTypeName == "Edm.Boolean")
+            {
+#>
+            return new EdmBooleanConstant(_<#= camelName #> ?? true);
```
[the remark this answers is not in the document]
Comment (xuzhg) [reply src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt]: @g2mula #resolved.

File: src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt (hunk 1 above)
Comment (xuzhg) [reply src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfigurationExtensions.tt]: #resolved

**Review by marabooy — APPROVED** [review]

**Review by g2mula — APPROVED** [review]
🐑

File: test/Microsoft.OData.ModelBuilder.Tests/PublicApi/Microsoft.OData.ModelBuilder.PublicApi.bsl [hunk 3]
```diff
@@ -1651,7 +1652,6 @@ public class Microsoft.OData.ModelBuilder.Capabilities.V1.AnnotationValuesInQuer
 
 	string TermName  { public virtual get; }
 
-	public Microsoft.OData.ModelBuilder.Capabilities.V1.AnnotationValuesInQuerySupportedConfiguration HasDynamicProperty (string name, object value)
```
Comment (mikepizzo) [inline test/Microsoft.OData.ModelBuilder.Tests/PublicApi/Microsoft.OData.ModelBuilder.PublicApi.bsl:1654]: @xuzhg -- Removing these will be a breaking change. Can we keep them, but mark as obsolete and remove in next major version?

**Pull request merged**